src/vm/callingconvention.h

```diff
@@ -1255,7 +1270,17 @@ int ArgIteratorTemplate<ARGITERATOR_BASE>::GetNextOffset()
         if (thValueType.IsHFA())
         {
             CorElementType type = thValueType.GetHFAType();
-            cFPRegs = (type == ELEMENT_TYPE_R4)? (argSize/sizeof(float)): (argSize/sizeof(double));
+            bool isSinglePrecision = (type == ELEMENT_TYPE_R4);
+
+            cFPRegs = isSinglePrecision ? (argSize/sizeof(float)): (argSize/sizeof(double));
+            m_hasArgLocDescForStructInRegs = true;
+
+            m_argLocDescForStructInRegs.Init();
+            m_argLocDescForStructInRegs.m_cGenReg = 0;
+            m_argLocDescForStructInRegs.m_cFloatReg = cFPRegs;
+            m_argLocDescForStructInRegs.m_idxGenReg = m_idxGenReg;
+            m_argLocDescForStructInRegs.m_idxFloatReg = m_idxFPReg;
+            m_argLocDescForStructInRegs.m_isSinglePrecision = isSinglePrecision;
```

Comment:
@rahku the reason why I have implemented it this way for unix amd64 is that there is no single offset in the TransitionBlock for structs passed in registers on Unix amd64. A struct can be split between upto two floating point and two general purpose registers. In this case, we return a special offset value TransitionBlock::StructInRegsOffset. I have originally refactored the code to let the GetNextOffset return a bitfield that would contain both the floating point registers and general purpose registers offsets, but it turned out to complicate code in many callers of the function and so it was decided during the code review to revert that change and rather use this side storage for these details.

But unless ARM64 can also split a single argument between multiple general purpose and floating point registers, I agree with you that we should use a single offset like we do e.g. for AMD64 on Windows.